specs/namespace/creation-for-whitelist.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the humbug/php-scoper package.
+ *
+ * Copyright (c) 2017 Théo FIDRY <theo.fidry@gmail.com>,
+ *                    Pádraic Brady <padraic.brady@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+return [
+    'meta' => [
+        'title' => 'Namespace declaration creation for whitelisted classes',
+        // Default values. If not specified will be the one used
+        'prefix' => 'Humbug',
+        'whitelist' => [
+        ],
+    ],
+
+    [
+        'spec' => <<<'SPEC'
+Single class should receive namespace
+SPEC
+        ,
+        'payload' => <<<'PHP'
+<?php
+
+class AppKernel
+{
+}
+
+----
+<?php
+
+namespace Humbug;
+
+class AppKernel
+{
+}
+
+PHP
+    ],
+
+    [
+        'spec' => <<<'SPEC'
```

Comment:
I would add two more cases:

- One with multiple different namespaces like you can find [here](https://github.com/humbug/php-scoper/blob/master/specs/class/regular.php#L145) (it doesn't need to be the same it's just for the idea)
- One with two classes belonging to the global namespace: one which is a whitelisted one, so `AppKernel` which as a result will be prefixed, and another which is not like `Closure` which should not be prefixed